File: internal/content/testdata/bench/java/nc-counting-bits.java
class Solution {
    public long[] counting_bits(long n) {
        long[] dp = new long[(int)n + 1];
        for (int i = 1; i <= (int)n; i++) {
            dp[i] = dp[i >> 1] + (i & 1);
        }
        return dp;
    }
}

File: internal/content/testdata/bench/rust/nc-counting-bits.rs
fn counting_bits(n: i64) -> Vec<i64> {
    let n = n as usize;
    let mut dp = vec![0i64; n + 1];
    for i in 1..=n {
        dp[i] = dp[i >> 1] + (i & 1) as i64;
    }
    dp
}
